adam_core/coordinates/tests/test_transforms_keplerian.py

```diff
@@ -0,0 +1,230 @@
+
+    # Assert positions are to within 10 mm
+    np.testing.assert_array_less(r_diff, 10)
+    # Assert velocities are to within 10 mm/s
+    np.testing.assert_array_less(v_diff, 10)
+
+
+def test_keplerian_to_cartesian_hyperbolic(orbital_elements):
+    # Test keplerian_to_cartesian (vmapped) against the expected cartesian elements
+    # of a series of sample orbital elements.
+
+    # Limit to hyperbolic orbits
+    orbital_elements = orbital_elements[orbital_elements["e"] > 1]
+
+    keplerian_elements = orbital_elements[["a", "e", "incl", "Omega", "w", "M"]].values
+    cartesian_elements_expected = orbital_elements[
+        ["x", "y", "z", "vx", "vy", "vz"]
+    ].values
+
+    cartesian_elements_actual = keplerian_to_cartesian(keplerian_elements)
+    diff = cartesian_elements_actual - cartesian_elements_expected
+
+    # Calculate offset in position in mm
+    r_diff = np.linalg.norm(diff[:, :3], axis=1) * u.au.to(u.mm)
+    # Calculate offset in velocity in mm/s
+    v_diff = np.linalg.norm(diff[:, 3:], axis=1) * (u.au / u.d).to(u.mm / u.s)
+
+    # Assert positions are to within 10 mm
+    np.testing.assert_array_less(r_diff, 10)
+    # Assert velocities are to within 10 mm/s
+    np.testing.assert_array_less(v_diff, 10)
+
+
+def test_cartesian_to_keplerian_elliptical(orbital_elements):
+    # Test cartesian_to_keplerian (vmapped) against the expected keplerian elements
+    # of a series of sample orbital elements.
+
+    # Limit to elliptical orbits
+    orbital_elements = orbital_elements[orbital_elements["e"] < 1]
+
+    keplerian_elements_expected = orbital_elements[
+        ["a", "q", "Q", "e", "incl", "Omega", "w", "M", "nu", "n", "P", "tp_mjd"]
+    ].values
+    epochs = orbital_elements["mjd_tdb"].values
+    cartesian_elements = orbital_elements[["x", "y", "z", "vx", "vy", "vz"]].values
+
+    # Cartesian to keplerian returns an N, 13 array containing the keplerian elements and cometary
+    # elements, with some additional columns.
+    keplerian_elements_actual = cartesian_to_keplerian(cartesian_elements, epochs)
+
+    # Remove semi-latus rectum
+    keplerian_elements_actual = keplerian_elements_actual[
+        :, [0, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+    ]
+
+    # Calculate the difference
+    diff = keplerian_elements_actual - keplerian_elements_expected
+
+    # Calculate offset in semi-major axis in m
+    a_diff = np.abs(diff[:, 0]) * u.au.to(u.m)
+    # Assert semi-major axis is to within 100 m
+    np.testing.assert_array_less(a_diff, 200)  # TODO
+
+    # Calculate offset in periapsis distance in m
+    q_diff = np.abs(diff[:, 1]) * u.au.to(u.m)
+    # Assert periapsis distance is to within 100 m
+    np.testing.assert_array_less(q_diff, 200)  # TODO
+
+    # Calculate offset in apoapsis distance in m
+    Q_diff = np.abs(diff[:, 2]) * u.au.to(u.m)
+    # Assert apoapsis distance is to within 100 m
+    np.testing.assert_array_less(Q_diff, 200)  # TODO
```

Comment:
Is this what you were saying about disagreement with horizon? Currently looks like we are testing within 200m?